src/rust/engine/src/externs.rs

```diff
@@ -145,6 +145,19 @@ pub fn project_multi(value: &Value, field: &str) -> Vec<Value> {
   })
 }
 
+pub fn project_bytes(value: &Value, field: &str) -> Vec<u8> {
+  let named_val = with_externs(|e| {
+    (e.project_ignoring_type)(
+      e.context,
+      value as &Handle,
+      field.as_ptr(),
+      field.len() as u64,
+    )
+    .into()
+  });
+  val_to_bytes(&named_val)
```

Comment:
I'm not sure how expensive `with_externs` is, but couldn't this be done in the same context? Something like:
```
with_externs(|e| {
  let named_val = ...;
  (e.val_to_bytes)(e.context, &named_val).to_bytes()
}
```

If I understand correctly, we are grabbing a read lock in `with_externs`, so we are not blocking additional readers, so it's not very important that we're doing two things inside a lock instead of one.

I'm probably in favour of the helper method if `with_externs` is cheap, as it seems it i, just checking that I'm not missing an important reason for the separate contexts.

Reply:
Inlined